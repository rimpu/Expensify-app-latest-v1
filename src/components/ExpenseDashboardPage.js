import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummaryConnected from '../components/ExpensesSummary'


const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummaryConnected />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
