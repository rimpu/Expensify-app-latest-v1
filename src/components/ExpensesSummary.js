import React from 'react'
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses'
import {getExpenses} from '../selectors/getExpenses'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component{
    render(){
        return(
            <p>Viewing {this.props.expenses.length} expense{this.props.expenses.length===1 ? '' :'s'} totalling {numeral(this.props.total / 100).format('$0,0.00')}</p>
        )
    }
} 

const mapStateToProps =(state)=>{
   return {
      expenses: selectExpenses(state.expenses, {...state.filters}),
      total : getExpenses(selectExpenses(state.expenses, {...state.filters}))
    }
}

export default connect(mapStateToProps)(ExpensesSummary)