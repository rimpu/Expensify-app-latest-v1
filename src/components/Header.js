import React from 'react';
import { NavLink } from 'react-router-dom';
import {connect} from 'react-redux'
import {LogOutUser} from '../actions/auth';

export const Header = (props) => {
  console.log(props)
  return (
    <header>
    <h1>Expensify</h1>
    <NavLink to="/dashboard" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    <button onClick={props.LogOutUser} >Log out</button>
    </header>
  )
 
  };

const mapDispatchToProps =(dispatch)=>{
  return {
    LogOutUser : ()=> dispatch(LogOutUser())
  }
}

export default connect(undefined,mapDispatchToProps)(Header);
