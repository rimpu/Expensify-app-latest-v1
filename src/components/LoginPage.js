import React from 'react'
import {connect} from 'react-redux'
import {AuthoriseUser} from '../actions/auth'

export const LoginButton = ({ AuthoriseUser }) => (
    <div>
      <button onClick={AuthoriseUser}>Login</button>
    </div>
  );
  
  const mapDispatchToProps = (dispatch) => ({
    AuthoriseUser: () => dispatch(AuthoriseUser())
  });

export default connect(undefined,mapDispatchToProps)(LoginButton)