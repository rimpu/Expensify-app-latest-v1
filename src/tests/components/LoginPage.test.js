import React from 'react';
import { shallow } from 'enzyme';
import {LoginButton} from '../../components/LoginPage'

test('Should render loginpage',()=>{
    const wrapper = shallow(<LoginButton AuthoriseUser= {()=>{}}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should call LoginButton ',()=>{
    const AuthoriseUser = jest.fn()
    const wrapper = shallow(<LoginButton AuthoriseUser= {AuthoriseUser}/>)
    wrapper.find('button').simulate('click')
    expect(AuthoriseUser).toHaveBeenCalled()
})