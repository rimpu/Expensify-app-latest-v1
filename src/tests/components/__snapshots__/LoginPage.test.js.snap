// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render loginpage 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
